stream_alert/rule_processor/classifier.py

```diff
@@ -212,50 +216,47 @@ def _check_valid_parse(self, valid_parses):
         Returns:
             [tuple] The proper tuple to use for parsing from the list of tuples
         """
-        if len(valid_parses) == 1:
+        # If there is only one parse or we do not have support for multiple schemas
+        # enabled, then just return the first parse that was valid
+        if len(valid_parses) == 1 or not SUPPORT_MULTIPLE_SCHEMA_MATCHING:
             return valid_parses[0]
 
         matched_parses = []
         for i, valid_parse in enumerate(valid_parses):
-            parser = valid_parse[1]
-            for data in valid_parse[2]:
-                if parser.matched_log_pattern(data, parser.options.get('log_patterns', {})):
-                    matched_parses.append(valid_parses[i])
-                    break
-                else:
-                    LOGGER.debug('log pattern matching failed for schema: %s', parser.schema)
+            log_patterns = valid_parse.parser.options.get('log_patterns', {})
+            if (all(valid_parse.parser.matched_log_pattern(data, log_patterns)
+                    for data in valid_parse.parsed_data)):
+                matched_parses.append(valid_parses[i])
+            else:
+                LOGGER.debug('log pattern matching failed for schema: %s', valid_parse.root_schema)
 
         if matched_parses:
             if len(matched_parses) > 1:
                 LOGGER.error('log patterns matched for multiple schemas: %s',
-                             ', '.join(name for name, _, _ in matched_parses))
-                LOGGER.error('proceeding with schema for: %s', matched_parses[0][0])
+                             ', '.join(parse.log_name for parse in matched_parses))
+                LOGGER.error('proceeding with schema for: %s', matched_parses[0].log_name)
 
             return matched_parses[0]
 
-
-
         LOGGER.error('log classification matched for multiple schemas: %s',
-                     ', '.join(name for name, _, _ in valid_parses))
-        LOGGER.error('proceeding with schema for: %s', valid_parses[0][0])
+                     ', '.join(parse.log_name for parse in valid_parses))
+        LOGGER.error('proceeding with schema for: %s', valid_parses[0].log_name)
 
         return valid_parses[0]
 
-    def _parse(self, payload, data):
-        """Parse a record into a declared type.
+    def _process_log_schemas(self, payload, data):
+        """Get any log schemas that properly classified this log
```

Comment:
The wording is weird. A schema doesn't classify a log, nor does a schema parse a log (said a bit below). We identify what schemas a log matches to.